### ISSUE TITLE
Do we need the "Video details" and the "Technical details" pages?

### ISSUE THREAD

**Issue by oas777** [elan-ev/tobira#807]
Everything that can be "done" (share) at

https://pr792.tobira.opencast.org/~manage/videos/O290gtPkcyw

can be "done" (better!) at

https://pr792.tobira.opencast.org/!v/O290gtPkcyw

and/or in the video editor (change metadata)

https://oc.tobira.ethz.ch/editor-ui/index.html?mediaPackageId=ad1890f6-07b9-4ad3-8346-60bef26e7a82.

And information from https://pr792.tobira.opencast.org/~manage/videos/O290gtPkcyw/technical-details can be added to the metadata.

**Comment by LukasKalbertodt** [comment]
You are right, but that's just a current technical limitation. In the first link you posted, users will be able to edit metadata once we implement that feature. (As this was one of the main thing we planned for this year.) Further, the share menu in the same page is indeed worse than the one on the actual video page. Once we nailed down the share menu design on the video page, we will take a look at this one again to improve it as well.

So I agree, currently this split and the whole "video details" page doesn't make too much sense. But this is already a preparation for the future. The technical details are most likely not interesting to most users, so putting it all into a separate sub page is a good idea in my opinion. Also remember that there will be more sub menus in the future: one for access rights (ACLs) for example.

**Comment by oas777** [comment]
https://pr792.tobira.opencast.org is offline, so we better revisit this when 2.0 is released and implemented. 

But: Functionalities should be available in one place only. So if the video editor offers an UI to edit metadata, there's no need for another page to do the same. If sharing is available on each video page, we don't need a "Videodetails" page to do the same.

**Comment by dagraf** [comment]
> But: Functionalities should be available in one place only. So if the video editor offers an UI to edit metadata, there's no need for another page to do the same. If sharing is available on each video page, we don't need a "Videodetails" page to do the same.

I disagree. I think, changing metadata should be as straightforward as possible and I would appreciate making it possible in the "Videodetails". Yes, metadata can be changed using the video editor. But it is not easy to find (and to explain to users) and needs quite a lot of clicks and waiting time until I managed to edit them (open video editor > go to metadata > find correct fields > edit > go to Fertig stellen > Does "Änderungen speichern" suffice or do I need to "Verarbeitung starten"?).

**Comment by LukasKalbertodt** [comment]
Re changing metadata, I agree with David. It's nice that the video editor can do it, but users would expect the workflow from YouTube, which is significantly easier. 
And actually, to my understanding, "changing metadata in Tobira" was one of the main next things we wanted to tackle. I thought we were all in agreement that this is one big missing feature right now and that it should work. Where did that communication go wrong? :thinking: 

> If sharing is available on each video page, we don't need a "Videodetails" page to do the same.

What if a video is not included in any page? Then there is no "video page". Then the only link that could be shared is the direct link, and that's what the "my videos" page offers. 
Further, while I generally agree with the sentiment of "functionality at one place", I don't think it needs to be that strict. We certainly shouldn't have two share menus with different UI. But if the same share button (with the same menu) would appear on the public video page and the "manage my videos" page, I don't think users would get confused, but instead appreciate the convenience.

**Comment by oas777** [comment]
- The fact that the (metadata) editor is "not easy to find" is something we could (and should) change, so that's only a pragmatic argument to have an additional place for metadata editing.
- The editor (like Studio) to me is part of Tobira, so I'm not speaking against "editing metadata in Tobira".
- "What if a video is not included in any page? Then there is no "video page"." Touché. And that's not an exotic use case. So we have to keep "video details". To at least edit the most important metadata (others in the editor). And a share menu like the one we have on the video details page.
- Still, I don't think we need "Technische Details".

**Comment by dagraf** [comment]
Only on the **subject of changing metadata**:
I agree with Lukas, that changing metadata directly in Tobira itself (and not in the video editor) is a big missing feature in order to make Tobira an easy-to-use video portal. Some more reasons/thoughts:
- Changing the metadata will probably be one of the most frequently used video editing options in Tobira. Therefore it deserves a prominent place in the portal and should be as straightforward (e.g., users should only see the relevant metadata and not all of them) and quick as possible (see bullet point below). 
- Improving the video (metadata) editor is an important thing to do, but ...
 - ... to change metadata using the video editor you need to hit "Verarbeitung starten" and then you have to wait a couple of minutes until reprocessing is over and the 'new' metadata end up being shown in Tobira. > This is a no-go for me.
 - ... it will probably take some time until the improvements are in place.
 - ... institutions may want to not offer the video editor to users but give them the opportunity to edit (some) metadata.

**Comment by oas777** [comment]
PS: Can we agree to get rid of the "Technical details" page?

**Comment by dagraf** [comment]
> PS: Can we agree to get rid of the "Technical details" page?

I would say Yes and No. Explanation:

- Yes: Most of the informations presented here are of no use as far as I think.
- But: The following information and possibilities should be kept and therefore moved to the "Videodetails page" (or somewhere else):
  - Opencast-ID
  - "Teil von:" > Suggestion: Write "Teil von Serie:" and not only offer the Series-ID but the clickable title of the Series (identical to the solution of the "Video page").
  - Option to download the video. Reason: Owners should be able to download their own videos regardless of whether download is enabled on these videos (I know that this feature is not yet implemented).

**Comment by oas777** [comment]
All of these could/should become part of the "Video details" page, I think. Actually, you wanted "dcterms:IsPartOf" ("Teil von:") to become editable anyway, makes sense to put it where the other "VIP metadata" (creator, title, description) can be edited.

**Comment by oas777** [comment]
I'm sure I didn't succeed in adding this to Q2, but I wanted to remind ourselves that we decided on May 24th that this issue (or rather "editing crucial metadata in Tobira" was supposed to be tackled "next". Just in case Ole is looking for work...

**Comment by oas777** [comment]
Ah!

**Comment by owi92** [comment]
Editing metadata in Tobira is tracked in #371, which is set for Q3. I don't think this could be done in the remaining weeks of Q2.
I will keep this issue here open for now, since it appears that we haven't come to a consensus whether to keep the "technical details" page or not. Maybe we can discuss this in our next meeting.

**Comment by oas777** [comment]
Fair enough. I just wanted to make sure it's on your radar. Let's table "technical details" for the next meeting, if that's not too late.

**Comment by owi92** [comment]
While we haven't had the chance to bring this up in the last meeting, after re-reading the comments here, let me try to summarize:
- The "Technical details" page will be removed
- In order to keep the informations presented there (at least the points @dagraf mentioned in his last comment) will be moved to the "Video details" page
- I would suggest that we have these in a separate list that is collapsed by default (see mockup below). We can of course change the placement of this, e.g. maybe put it all the way to the bottom. (the navigation menu on the left would be changed as well of course)

<img width="510" alt="Bildschirmfoto 2023-07-05 um 15 17 12" src="https://github.com/elan-ev/tobira/assets/94838646/b96889a0-93d6-41ad-a2a3-c75b1d772cd1">
<img width="510" alt="Bildschirmfoto 2023-07-05 um 15 36 45" src="https://github.com/elan-ev/tobira/assets/94838646/c2db5fbc-e082-4d33-9894-75444722f605">

**Comment by LukasKalbertodt** [comment]
The collapsable technical details are not a bad idea. But I just mentioned in our meeting: I'd propose to halt this discussion and only talk about this again when we can actually edit metadata and ACLs. I think we are discussing this in a total vacuum. Let's first  actually implement useful features and once that is done, we can still decide how to best separate them in different pages (or not). Olaf, Ole and Julian seemed to (even if reluctantly :P) agree.

And no, this is not (just :grimacing:) a way to keep you from deleting my beloved technical details page. I really think the discussion makes little sense right now. And I wish I had thought of bringing this up 2 months ago.

**Comment by oas777** [comment]
Just a quick shout-out this hasn't been forgotten... and even with "Access policy" and "Statistics" upcoming, I don't foresee a bright future for the "Technical details" (which currently doesn't hold any relevant information we don't have somewhere else).

**Comment by LukasKalbertodt** [comment]
I created #1392 to focus this discussion and combine some issues. I will close this as it contains some outdated discussions. But no worries: this issue is linked from the new one.
